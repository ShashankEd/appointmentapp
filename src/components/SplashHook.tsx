import React, {useEffect, useState } from 'react';
import {
    View,
    Text
} from 'react-native';
import { useSelector } from "react-redux";
interface props {
    navigation: {
        navigate: (routeName: string) => void;
    };
    state: {
        storeWhetherUserLoggedIn: {
            flag: boolean
        }
    }
}
const SplashHook: React.FC<props> = (props) =>{
    const [navigation, setNavigation] = useState(props.navigation);
    useEffect (() => {
        setTimeout(() => {
            navigation.navigate('HomeHook');
        },2000)
    },[])
    return (
        <View style={{ flex: 1, padding: 24,backgroundColor:'yellow'}}>
            <Text>Splash</Text>
        </View>
    )
}

export default SplashHook;