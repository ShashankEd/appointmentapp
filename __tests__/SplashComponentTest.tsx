import 'react-native';
import React from 'react';
import SplashHook from '../src/components/SplashHook';
import renderer, {create} from 'react-test-renderer'

//if any asynchronous task is running in the comopnent, then use it
jest.useFakeTimers();
//@ts-ignore
const tree = create(<SplashHook/>);
test('snapshot',() => {
    expect(tree).toMatchSnapshot()
});



