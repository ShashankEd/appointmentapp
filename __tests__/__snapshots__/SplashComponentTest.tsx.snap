// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DDDDDD",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Verdana",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Appointment Booking App
  </Text>
</View>
`;
